com.spotify.Client.desktop

```diff
@@ -0,0 +1,13 @@
+[Desktop Entry]
```

Comment:
I really don't like this, its unclear what we even change.

Reply:
It works at least? I tried ```desktop-file-edit``` every which way I could think of and I could not get it to work?

Reply:
Thats not a very good excuse, you could just figure out what was going wrong.

Reply:
I can only get ```desktop-file-edit``` to actually do anything inside of ```build-commands``` and from there I don't know where the upstream .desktop is or if it even exists yet? ```desktop-file-edit``` doesn't do anything in ```commands``` and fails silently from what I can tell.

If I could get a hold of the upstream .desktop in ```build-commands``` I would just use ```desktop-file-edit``` to change the upstream .desktop but really it's 6 of 1 thing 1/2 a dozen of another. The end result would be the same.

Reply:
As far as what was changed:

Type=Application
Name=Spotify
GenericName=Music Player    >>> Online music streaming service (better generic name IMHO)
Comment=Access all of your favorite music    <<< added. (Blurp for whoever still uses tooltips?)
Icon=spotify-client    >>> com.spotify.Client (New icon)
TryExec=spotify
Exec=spotify %U
Terminal=false
MimeType=x-scheme-handler/spotify;
Categories=Audio;Music;Player;AudioVideo;     >>> Audio;Music;AudioVideo; (Player is not a valid Category. neither is Music but people still use it a lot)
Keywords=Music;Player;Streaming;Online;    <<< added (for searching)
StartupWMClass=spotify    >>> Spotify (I'm not sure why? It's the way you had it)

Reply:
Can `desktop-file-edit` work for `extra-data`? Nevertheless shipping own desktop file only to change some description is over the top for me.

Reply:
This is just one of many steps to fixing several papercuts that I've noticed using Sootify and reading issues here and on the Spotify forum.

We may not be able to tinker with it's internals but we can control how it presents it self to the desktop environment at least to a small degree.

If we can improve the experience even in a small way I think we should.

Reply:
If there were a way to upstream these improvements I would, but Spotify doesn't have a way to do that. They don't even have someone who's in charge of the Linux version full time. The person who does it does it on their free time an on hack days.

Reply:
Besides we already ship a custom appstream. I'm not sure how this is any more sacrilegious.

Reply:
You won't be able to build flatpak without proper appstream file so this isn't a choice but necessity.

I agree about fixing technical glitches like correct icon for pulseaudio which was flatpak specific issue but modifying things like app descriptions or icons which will differ from official deb or snap packages seems out of place to me.

Reply:
It would be a different story if Spotify had their crap together and actually actively developed and took some time on the Linux version but they don't. They're not curating an experience. Their .desktop file is a clear example of doing just enough to make it technically functional but no more copypasta.

Reply:
In this case we're not stepping on their toes because they don't have any.

Reply:
When/if they ever decide to actually care about the experience on Linux and include more than just the bare minimum then we can use what provide.

Reply:
My understanding for flathub role is different - it's just publishing platform that serves as a bridge between upstream and users. Ideally spotify would maintain it by themselves same as snap but if they aren't then downstream modifications should be minimized to retain app functionality in flatpak environment. For me the app should look and behave the same way as when installed from deb or snap.

Spotify has no obligation to care about linux users experience. It's their proprietary code and their choice. Even if it's crap it's their crap and their responsibility.

Reply:
To be clear: this is just my opinion and I'm not maintainer of this flatpak. This repo already contains things which I wouldn't accept so it's plausible that maintainer opinion is different.

Reply:
I don't even think they maintain the snap. I'm pretty sure it's just endorsed by them.

Reply:
> downstream modifications should be minimized to retain app functionality in flatpak environment.

This doesn't do anything to change app functionality. There's a big difference between presentation and functionality.

Reply:
They're throwing out an app with A LOT of rough edges. I look at this turd polishing as doing them a favor.

Reply:
My hope is that someone at Spotify might notice that the flatpak experience is better then the upstream package's experience and they'll either do a little work to improve their experience and/or adopt some of the changes upstream. In any event even if they don't I don't see why we have to settle for subpar if we can do something about it.

Reply:
Honestly most of those changes are matter of taste rather than actual improvement. I preferred old icon than new one which is darker so it's not like you always improve someone experience. I don't expect 99% users would even care about changes done here (as user I wouldn't).

Reply:
> Honestly most of those changes are matter of taste rather than actual improvement.

Maybe as in *a* taste as opposed to absolutely no taste specified .

> I preferred old icon than new one which is darker

Then the old icon doesn't comply to there own branding documentation because I used the exact colors they specify.

> I don't expect 99% users would even care about changes done here

That maybe true but all the little changes have a cumulative effect.

Reply:
>I can only get desktop-file-edit to actually do anything inside of build-commands and from there I don't know where the upstream .desktop is or if it even exists yet? desktop-file-edit doesn't do anything in commands and fails silently from what I can tell.

>If I could get a hold of the upstream .desktop in build-commands I would just use desktop-file-edit to change the upstream .desktop but really it's 6 of 1 thing 1/2 a dozen of another. The end result would be the same.

So yay or nay on this @TingPing? Your thoughts?